coderd/users.go

```diff
@@ -764,6 +761,52 @@ func (api *API) putUserStatus(status database.UserStatus) func(rw http.ResponseW
 	}
 }
 
+// @Summary Update user's theme preference
+// @ID update-user-theme-preference
+// @Security CoderSessionToken
+// @Accept json
+// @Produce json
+// @Tags Users
+// @Param user path string true "User ID, name, or me"
+// @Param request body codersdk.UpdateUserThemePreferenceRequest true "New theme preference"
+// @Success 200 {object} codersdk.User
+// @Router /users/{user}/theme [put]
+func (api *API) putUserThemePreference(rw http.ResponseWriter, r *http.Request) {
+	var (
+		ctx  = r.Context()
+		user = httpmw.UserParam(r)
+	)
+
+	var params codersdk.UpdateUserThemePreferenceRequest
+	if !httpapi.Read(ctx, rw, r, &params) {
+		return
+	}
+
+	updatedUser, err := api.Database.UpdateUserThemePreference(ctx, database.UpdateUserThemePreferenceParams{
+		ID:              user.ID,
+		ThemePreference: params.ThemePreference,
+		UpdatedAt:       dbtime.Now(),
+	})
+	if err != nil {
+		httpapi.Write(ctx, rw, http.StatusInternalServerError, codersdk.Response{
```

Comment:
I _would want_ us to check `database.IsQueryCanceledError` here and not log an internal server error, but there's soo many places we don't so please pay me no mind and leave it as-is. 😅 (This is just informative.)

Reply:
yeah, most of the backend is copy-pasted from the `UpdateUserProfile` stuff. 😅 I'll let someone more knowledgable about this bit of the codebase clean it up one day.